backend/src/main/kotlin/com/github/quillraven/toilapp_dataloader/DataLoaderApplication.kt

```diff
@@ -0,0 +1,144 @@
+            "com.github.quillraven.toilapp_dataloader"]
+)
+class DataLoaderApplication
+
+@Component
+class DataLoaderRunner(
+        @Autowired private val imgService: ImageService,
+        @Autowired private val toiletService: ToiletService,
+        @Autowired private val userService: UserService,
+        @Autowired private val context: ConfigurableApplicationContext
+): CommandLineRunner {
+
+    val numToilets = 20
+
+    override fun run(vararg args: String?) {
+        createUsers().flatMap { createToilets(it) }.subscribe{
+            println("$numToilets toilets created..")
+            exitProcess(SpringApplication.exit(context))
+        }
+    }
+
+    private fun createUsers(): Mono<List<Any>> {
+        val userMonoList = mutableListOf<Mono<User>>()
+        //create users
+        for(usrName in userNames) {
+            val userName = userNames[Random.nextInt(0, userNames.size)]
+            val email = "$userName@mail.com"
+            val user = User(UUID.randomUUID().toString(), userName, email)
+            userMonoList.add(userService.createUser(user))
+        }
+        return Mono.zip(userMonoList){ uArr -> uArr.toList() }
+    }
+
+    private fun createToilets(uList: List<Any>): Mono<List<Any>> {
+        val toiletMonoList = mutableListOf<Mono<Toilet>>()
+        for(num in 0 until numToilets) {
+            toiletMonoList.add(createToilet(uList))
+        }
+        return Mono.zip(toiletMonoList){ tArr -> tArr.toList()}
+    }
+
+    private fun createToilet(uList: List<Any>): Mono<Toilet> {
+        val imgName = "toilet" +  Random.nextInt(1, 10) + ".jpg"
+        val title = titles[Random.nextInt(0, titles.size)]
+        val description = descriptions[Random.nextInt(0, descriptions.size)]
+        val rating = Random.nextDouble(1.0, 6.0)
+        val numCommands = Random.nextInt(0, 4)
+        val comments = mutableListOf<Comment>()
+        val log = Random.nextDouble(10.0, 20.0)
+        val lat = Random.nextDouble(45.0, 55.0)
+
+        val userList = mutableListOf<User>()
+        userList.addAll(uList as List<User>)
+        for(i in 0..numCommands) {
+            val user = userList.get(Random.nextInt(0, userList.size))
+            val comment = Comment(UUID.randomUUID().toString(), user, Date(), commentTexts[Random.nextInt(0, commentTexts.size)])
+            comments.add(comment)
+        }
+
+        val inStreamCallable: Callable<InputStream> = Callable<InputStream>{ DataLoaderRunner::class.java.getResourceAsStream("/sample-images/$imgName")}
+        val objId = imgService.store(inStreamCallable, "my-name")
+        return objId.flatMap{oid -> run{
+            println("imgOid = $oid")
+            val toilet = Toilet(UUID.randomUUID().toString(), title, GeoJsonPoint(log, lat), oid.toHexString(), rating,
+                    false, false, description, comments.toTypedArray(), emptyArray())
```

Comment:
Kotlin provides named parameters and therefore we can name the two `false` parameters to make it clear:

```kotlin
val toilet = Toilet(
                UUID.randomUUID().toString(),
                title,
                GeoJsonPoint(log, lat),
                oid.toHexString(),
                rating,
                disabled = false,
                toiletCrewApproved = false,
                description,
                comments.toTypedArray(),
                emptyArray()
            )
```

I know it is just a data loader and it is not super important but I just want to point it out in case you are not aware of this feature ;)